ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestination.java

```diff
@@ -293,21 +306,18 @@ private boolean updateBlobMetadata(BlobId blobId, Map<String, Object> updateFiel
           // attempt fails.  So we check for that case here.
           CloudBlobMetadata cosmosMetadata = cosmosDataAccessor.getMetadataOrNull(blobId);
           if (cosmosMetadata != null) {
-            if (cosmosMetadata.isDeletedOrExpired()) {
+            if (cosmosMetadata.isCompactionCandidate(
+                TimeUnit.HOURS.toMillis(cloudConfig.cloudBlobCompactionIntervalHours))) {
               logger.warn("Inconsistency: Cosmos contains record for inactive blob {}, removing it.", blobId.getID());
               cosmosDataAccessor.deleteMetadata(cosmosMetadata);
               azureMetrics.blobUpdateRecoverCount.inc();
```

Comment:
It seems like the code will now throw in this case. Is there a reason for the change?

Reply:
Thanks for the explanation. I think that makes sense